Zend/zend_execute.c

```diff
@@ -721,6 +721,7 @@ static zend_never_inline ZEND_COLD int ZEND_FASTCALL make_real_object_rw(zval *o
 		return 0;
 	}
 	object_init(object);
+	zend_error(E_WARNING, "Creating default object from empty value");
```

Comment:
It would be better to replace the use of make_real_object_rw with make_real_object instead. If you look at that function it has a bunch of special handling to avoid some edge-case memory safety issues, and it's best if we can just reuse it.